### ISSUE TITLE
Condense required rbac roles for keda-operator

### ISSUE THREAD

**Issue by anggao** [kedacore/keda#1020]
## Expected Behavior
https://github.com/kedacore/keda/blob/master/deploy/10-cluster_role.yaml 
clusterrole should only list the required roles keda-operator needs to operate

## Actual Behavior
keda-operator requires full control of pods or secrets etc.

## Specifications

  - **KEDA Version: v.1.5.0
  - **Platform & Version: N/A
  - **Kubernetes Version: v1.17.5
  - **Scaler(s): N/A

**Comment by anggao** [comment]
The following rbac roles seems enough

```
- apiGroups:
  - ""
  resources:
  - pods
  - services
  - services/finalizers
  - endpoints
  - events
  - configmaps
  - namespaces
  - external
  verbs:
  - get
  - list
- apiGroups:
  - ""
  resources:
  - configmaps
  verbs:
  - '*'
```

**Comment by zroubalik** [comment]
Thanks, good catch. The verbs for the pods/secrets etc should contain `watch` as well.

**Comment by zroubalik** [comment]
#fixed by #1024
